tests/plugins/autogrow/autogrow.js

```diff
@@ -8,7 +8,58 @@
 
 	bender.editor = {};
 
+	bender.createEditor = function( width, paragraphs ) {
+		bender.editorBot.create( {
+			name: 'EditorWidth' + width,
+			config: {
+				width: width
+			}
+		}, function( bot ) {
+			bot.setData( autogrowTools.getTestContent( paragraphs ), function() {
+				var editor = bot.editor;
+				var initialEditorSize = autogrowTools.getEditorSize( editor );
+
+				editor.once( 'afterCommandExec', function() {
+					resume( function name( params ) {
+						var editorSize = autogrowTools.getEditorSize( editor );
+
+						assert.isTrue( editorSize.height > initialEditorSize.height, 'editor height should increase' );
+						assert.areEqual( editorSize.width, initialEditorSize.width, 'editor width should not change' );
+					} );
+				} );
+
+				editor.execCommand( 'autogrow' );
+
+				wait();
+			} );
+		} );
+	};
+
 	bender.test( {
+		// (#4372)
+		'test autogrow for editor width 200%': function() {
+			bender.createEditor( '200%', 6 );
+		},
+		// (#4372)
+		'test autogrow for editor width 20em': function() {
+			bender.createEditor( '20em', 6 );
+		},
+		// (#4372)
+		'test autogrow for editor width 200px': function() {
+			bender.createEditor( '200px', 6 );
+		},
+		// (#4372)
+		'test autogrow for editor width 200': function() {
+			bender.createEditor( 200, 6 );
+		},
+		// (#4372)
+		'test autogrow for editor width 0': function() {
+			bender.createEditor( 0, 6 );
+		},
+		// (#4372)
+		'test autogrow for editor width auto': function() {
+			bender.createEditor( 'auto', 6 );
+		},
```

Comment:
Since we are not changing the number of paragraphs anywhere, maybe let's just hardcode it inside the test method? We could also choose a bit higher value to make sure to not depend on some edge case, like 10.